navigation/navigation-common/src/commonMain/kotlin/androidx/navigation/NavDeepLink.kt

```diff
@@ -727,13 +728,13 @@ internal constructor(
         val uriRegex = StringBuilder("^")
         // append scheme pattern
         if (!SCHEME_PATTERN.containsMatchIn(uriPattern)) {
-            uriRegex.append("http[s]?://")
+            uriRegex.append(SCHEME_REGEX)
         }
         // extract beginning of uriPattern until it hits either a query(?), a framgment(#), or
         // end of uriPattern
         Regex("(\\?|#|$)").find(uriPattern)?.let {
             buildRegex(uriPattern.substring(0, it.range.first), pathArgs, uriRegex)
-            isExactDeepLink = !uriRegex.contains(".*") && !uriRegex.contains("([^/]+?)")
+            isExactDeepLink = !uriRegex.contains(WILDCARD_REGEX) && !uriRegex.contains(PATH_REGEX)
```

Comment:
PATH_REGEX != `"([^/]+?)"`

Reply:
that's why there is a comment: `allows for empty path arguments i.e. empty strings ""`